src/main/java/org/mockito/internal/framework/DefaultMockitoSession.java

```diff
@@ -0,0 +1,44 @@
+package org.mockito.internal.framework;
+
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.MockitoSession;
+import org.mockito.exceptions.misusing.RedundantListenerException;
+import org.mockito.internal.exceptions.Reporter;
+import org.mockito.internal.junit.TestFinishedEvent;
+import org.mockito.internal.junit.UniversalTestListener;
+import org.mockito.internal.util.MockitoLogger;
+import org.mockito.quality.Strictness;
+
+public class DefaultMockitoSession implements MockitoSession {
+
+    private final Object testClassInstance;
+    private final UniversalTestListener listener;
+
+    public DefaultMockitoSession(Object testClassInstance, Strictness strictness, MockitoLogger logger) {
+        this.testClassInstance = testClassInstance;
+        listener = new UniversalTestListener(strictness, logger);
+        try {
+            Mockito.framework().addListener(listener);
```

Comment:
Why add a listener here? It seems that the code is trying to detect invalid usage of mocking, but in `finishMocking` this listener is automatically removed and manually invoked with `testFinished`. This seems like a misusage of `Mockito.framework()`